virtual/vbox/vbox_test.go

```diff
@@ -53,20 +54,44 @@ func TestVmBase(t *testing.T) {
 		t.Fatalf("expected virtual machine to have been added")
 	}
 
-	err = vm.Start(ctx)
+	teststart := func() {
+		err = vm.Start(ctx)
+		if err != nil {
+			t.Fatalf("unexpected error when starting vm: %s", err)
+		}
+
+		output, err = execute("list", "runningvms")
+		if err != nil {
+			t.Fatalf("unexpected error when listing running vms: %s", err)
+		}
+
+		if !strings.Contains(output, name) {
+			t.Fatalf("expected virtual machine to be running")
+		}
+	}
+	teststart()
+
+	err = vm.Suspend(ctx)
 	if err != nil {
-		t.Fatalf("unexpected error when starting vm: %s", err)
+		t.Fatalf("unexpected error when suspending vm: %s", err)
 	}
 
 	output, err = execute("list", "runningvms")
 	if err != nil {
 		t.Fatalf("unexpected error when listing running vms: %s", err)
 	}
 
-	if !strings.Contains(output, name) {
-		t.Fatalf("expected virtual machine to be running")
+	if strings.Contains(output, name) {
+		t.Fatalf("expected virtual machine to have been stopped")
 	}
 
+	// Check state using info command
+	if info := vm.Info(); info.State != virtual.Suspended {
+		t.Fatalf("expected virtual machine to be suspended")
+	}
+
+	teststart()
+
```

Comment:
Probably, this will gonna work, however  when there is a interaction between vboxmanager and OS on waking up an machine which is  in saved state, causes problems, it is unable to wake the machine up correctly. It is something related to vboxmanage and OS, not from your code.

Reply:
Hmm didn't experience that.
Will try it again to be sure i didn't miss something.